src/gui/sdlmain.cpp

```diff
@@ -3094,10 +3144,21 @@ void GFX_LosingFocus()
 	MAPPER_LosingFocus();
 }
 
-bool GFX_IsFullscreen(void) {
+bool GFX_IsFullscreen() {
 	return sdl.desktop.fullscreen;
 }
 
+void GFX_RegenerateWindow(Section *sec) {
+	if (!first_window) {
+		const auto section = static_cast<const Section_prop *>(sec);
+		if (strcmp(section->Get_string("output"), "surface"))
+			remove_window();
+		set_output(sec, wants_stretched_pixels());
+		GFX_ResetScreen();
+	}
+	first_window = false;
```

Comment:
(Very minor/optional suggestion to drop a nesting level and avoid redundant assignment):
``` c++
	if (first_window) {
		first_window = false;
		return;
	}
	const auto section = static_cast<const Section_prop *>(sec);
	if (strcmp(section->Get_string("output"), "surface"))
		remove_window();
	set_output(sec, wants_stretched_pixels());
	GFX_ResetScreen();
```

Reply:
Done accordingly.